### ISSUE TITLE
Blazor WASM: parameterset called after navigation reuses querystring value from previous page even if the new page's url has no querystring parameters

### ISSUE THREAD

**Issue by luisabreu** [dotnet/aspnetcore#53364]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Hello.

We're seing a strange behavior in our Blazor WASM app when a user navigates from one page to another and both pages have parameters which are fed from the querystring and both qs parameters have the same name. Here's what's happening:

1. User loads initial page and performs a simple search (simulated on page1 of the sample) by filling the textbox and clicking the search button (the textbox is bound to a model managed by a ViewModel which is injected through DI on the page)
2. When the search button is clicked, the uri of the page is updated so that the it reflects the current search. This is done by using the NavigationService and calling NavigateTo. For insance, if the user entered the value 100, the uri will be set to"page1?id=100".
3. Navigating to the same page fires the parameterset event for page1, which, in this case, doesn't do anything because in our app it will only be handled once during the first time that the  page is loaded (theres a flag which gets set to true after being run for the 1st time)
4. if the user navigates to page2 after the previous search, and even though the uri shown by the browser is http://localhost:5001/page2 only, the parameterset event of page 2 will be called twice. During the 1st call, page2's parameter is set to 100 (last value passed to the query string of page1 after running the search) and then it's called again with that parameter set to null
5. Since we only handle the 1st parameterset event (like page1, during the second parameterset event, we don't do anything), then this second page's property will end up being set to 100 value even though there's no value  for the  querystring parameter that is used to set it up.

Here's the sequence with images (regarding the console, Page info references the page's parameter value that is fed from the querystring and page model reflects the  value that is bound to the textbox - this value is initialized during parameterset event by copying the page's Info property):

![Initial load of the page](https://github.com/dotnet/aspnetcore/assets/521086/d621677d-2483-411d-aba4-bd185cce9351)

Setting the textbox to 100 and clicking the button updates the page's url:

![Setting the textbox to 100 and clicking the button](https://github.com/dotnet/aspnetcore/assets/521086/244c647b-2bf9-4397-8e07-6f522716fe27)

As you can see, parameterset event is fired, and both values are in sync (page's property is updated from the query string and the mode's Info property was updated through the binding that connects it to the textbox). Since the page is the same, the control gets reused and everything works properly.

When we navigate to page2, which also has a property that is filled from a qs parameter with the same name as the one we had in page1, then we  model is initialized  with the previous qs value that was applied to page1:

![Navigating to page2](https://github.com/dotnet/aspnetcore/assets/521086/921b830f-10a3-482f-8bd6-dfb9fb7e5a90)

Btw, here are 2 captures of the call stacks from within parameterset event handled by page2:

![call stack from within parameterset 1](https://github.com/dotnet/aspnetcore/assets/521086/d57a8981-94aa-44ac-848e-317580ec67d9)

![call stack from within parameterset 2](https://github.com/dotnet/aspnetcore/assets/521086/fec66077-61f8-4432-988e-6ba689887ae6)

In the real world app, all the pages of the handle parameterset so that only the 1st call is handled for initializing the model that is bound to the ui of each page. When the user runs a search, the page call a rest service, get the values and then update the URI by navigating to the same page (we're only updating the page's uri so that searches can be shared between users). We've ended up with this approach (handling parameterset only once during its first call)  because the pages are rather complex and it seemed to work properly until now.  

Is this a bug? Should parameterset be fired 2 times when navigating from page1 to page2? And if this is the expected behavior (btw, can someone tell me where it's described on the docs) why do I get the value 100 during that first call if the there's no querystring values on the url that ends up loading the page?

Thanks.

### Expected Behavior

I expected to see parameterset event fired once only after navigating to page2 and since there's no querystring parameters defined, the second page's property that gets its value from the querystring  shouldn't have been set to the value that existed in the previous uri.

### Steps To Reproduce

There's sample code that reproduces the problem (as described above) here: https://github.com/luisabreu/DemoQueryStringParameterIssue

### .NET Version

8.0.101

### Anything else?

here's the log file for the steps described above when run on my machine:
e
[error.log](https://github.com/dotnet/aspnetcore/files/13931943/error.log)

**Comment by luisabreu** [comment]
Any feedback?

**Comment by luisabreu** [comment]
Seems to be this one #44781

**Comment by MackinnonBuck** [comment]
Thanks for reaching out. This seems to be a duplicate of https://github.com/dotnet/aspnetcore/issues/52483. This is going to be addressed in an upcoming .NET 8 servicing release, and this fix has already been merged, so I'm going to close out this issue.
